src/variety/kuroclone.js

```diff
@@ -0,0 +1,246 @@
+			edit: ["border", "number", "direc", "clear"],
+			play: ["shade", "unshade"]
+		},
+		mouseinput_auto: function() {
+			if (this.puzzle.playmode) {
+				if (this.mousestart || this.mousemove) {
+					this.inputcell();
+				}
+			} else if (this.puzzle.editmode) {
+				if (this.mousestart || this.mousemove) {
+					if (this.isBorderMode()) {
+						this.inputborder();
+					} else {
+						this.inputdirec();
+					}
+				} else if (this.mouseend && this.notInputted()) {
+					this.inputqnum();
+				}
+			}
+		}
+	},
+
+	//---------------------------------------------------------
+	// キーボード入力系
+	KeyEvent: {
+		enablemake: true,
+		moveTarget: function(ca) {
+			if (ca.match(/shift/)) {
+				return false;
+			}
+			return this.moveTCell(ca);
+		},
+
+		keyinput: function(ca) {
+			if (this.key_inputdirec(ca)) {
+				return;
+			}
+			this.key_inputqnum(ca);
+		}
+	},
+
+	//---------------------------------------------------------
+	// 盤面管理系
+	Cell: {
+		numberRemainsUnshaded: true,
+		minmum: 1,
+		maxnum: 99
+	},
+	Board: {
+		hasborder: 1
+	},
+	BoardExec: {
+		adjustBoardData: function(key, d) {
+			this.adjustNumberArrow(key, d);
+		}
+	},
+	AreaRoomGraph: {
+		enabled: true,
+		hastop: false
+	},
+	AreaShadeGraph: {
+		enabled: true
+	},
+
+	//---------------------------------------------------------
+	// 画像表示系
+	Graphic: {
+		gridcolor_type: "LIGHT",
+
+		enablebcolor: true,
+		bgcellcolor_func: "qsub1",
+
+		paint: function() {
+			this.drawBGCells();
+			this.drawGrid();
+			this.drawShadedCells();
+			this.drawArrowNumbers();
+			this.drawBorders();
+			this.drawChassis();
+			this.drawBoxBorders(false);
+			this.drawTarget();
+		}
+	},
+
+	//---------------------------------------------------------
+	// URLエンコード/デコード処理
+	Encode: {
+		decodePzpr: function(type) {
+			this.decodeBorder();
+			this.decodeArrowNumber16();
+		},
+		encodePzpr: function(type) {
+			this.encodeBorder();
+			this.encodeArrowNumber16();
+		}
+	},
+	//---------------------------------------------------------
+	FileIO: {
+		decodeData: function() {
+			this.decodeAreaRoom();
+			this.decodeCellDirecQnum();
+			this.decodeCellAns();
+		},
+		encodeData: function() {
+			this.encodeAreaRoom();
+			this.encodeCellDirecQnum();
+			this.encodeCellAns();
+		}
+	},
+
+	//---------------------------------------------------------
+	// 正解判定処理実行部
+	AnsCheck: {
+		checklist: [
+			"checkSideAreaShadeCell",
+			"checkUnitExists",
+			"checkArrowNumber",
+			"checkUnitsCount",
+			"checkUnitsShape",
+			"checkNumberHasArrow"
+		],
+
+		// From shimaguni.js
+		checkSideAreaShadeCell: function() {
+			this.checkSideAreaCell(
+				function(cell1, cell2) {
+					return cell1.isShade() && cell2.isShade();
+				},
+				false,
+				"cbShade"
+			);
+		},
+
+		// Check unit exists toward arrow
+		checkUnitExists: function() {
+			var bd = this.board;
+			for (var c = 0; c < bd.cell.length; c++) {
+				var cell = bd.cell[c];
+				if (!cell.isValidNum() || cell.qdir === 0) {
```

Comment:
This check currently doesn't work for hatenas with arrows:

https://pzprjs-git-fork-subaru-saito1-kuroclone-robx.vercel.app/p.html?type=editor&pzprv3/kuroclone/2/3/1/0%200%200%20/0%200%200%20/.%203%2C-%20.%20/.%20.%20.%20/.%20.%20.%20/%23%20.%20%23%20//

Reply:
It's actually possible for a cell to have qdir but not qnum, which currently causes an error:
1. Place a number
2. Draw an arrow on the number
3. Erase the number with backspace
4. Check for errors. This will cause an empty cell to light up.

If you replace the check `cell.isValidNum()` with `cell.isNum()` (instead of removing it) you can check for question marks.

Reply:
Thanks for the info. I fixed it!